model/attachmentFilter.php

```diff
@@ -0,0 +1,17 @@
+<?php
+require_once "filterDecorator.php";
+
+class AttachmentFilter extends FilterDecorator
+{
+    public $ticket;
+    public function __construct($filter)
+    {
+        $this->filter = $filter;
+    }
+
+    public function generate()
+    {
+          return  $this->filter->generate(). "SELECT * FROM tickets WHERE TRUE AND EXISTS (SELECT * FROM attachment WHERE Trueattachment.ticket_id='{$this->ticket->id}')";

```

Comment:
....
Recall that `$this->filter->generate()` already generates the `SELECT * FROM tickets WHERE TRUE` part, so remove everything before the ` AND`.